src/spikeinterface/extractors/neoextractors/neobaseextractor.py

```diff
@@ -251,40 +254,40 @@ def get_traces(
 
 
 class NeoBaseSortingExtractor(_NeoBaseExtractor, BaseSorting):
-    # this will depend on each reader
-    handle_spike_frame_directly = True
+    neo_returns_timestamps = False
+    # `neo_returns_timestamps` is a class attribute indicating whether
+    # `neo_reader.get_spike_timestamps` returns timestamps instead of frames,
+    # If True, then the segments do a transformation to frames.
+    # For formats that return timestamps (e.g. Mearec, blackrock) this should be set to
+    # True in the format class that inherits from this.
+
+    need_t_start_from_signal_stream = False
+    # `need_t_start_from_signal_stream` is a class attribute indicating whether t_start should be inferred
+    # from `neo_reader.get_signal_t_start`. If True, then we try to infer t_start from the signal stream with
+    # corresponding sampling frequency. If False, then t_start is set to None which is 0 for practical purposes.
 
     def __init__(self, block_index=None, sampling_frequency=None, use_natural_unit_ids=False, **neo_kwargs):
         _NeoBaseExtractor.__init__(self, block_index, **neo_kwargs)
 
         self.use_natural_unit_ids = use_natural_unit_ids
-
-        # Get the stream index corresponding to the extracted frequency
         spike_channels = self.neo_reader.header["spike_channels"]
-
         if use_natural_unit_ids:
             unit_ids = spike_channels["id"]
             assert np.unique(unit_ids).size == unit_ids.size, "unit_ids is have duplications"
         else:
             # use interger based unit_ids
-            unit_ids = np.arange(spike_channels.size, dtype="int64")
+            unit_ids = np.arange(spike_channels.size, dtype="int32")
```

Comment:
One day when we will have more than 2G units we will regret that choice :)

Reply:
x D